src/pybind/mgr/orchestrator.py

```diff
@@ -384,36 +386,158 @@ class DriveGroupSpec(object):
     """
     Describe a drive group in the same form that ceph-volume
     understands.
+    This structure allows the selection of storage devices
+    in a set of nodes
     """
-    def __init__(self, devices):
-        self.devices = devices
+    def _init_attribute(self, att_dict, attribute_name, default_value):
+        """ Extract attribute from dict, if not present applies default value
+
+        :param att_dict      : dictionary with attributes
+        :param attribute_name: The attribute to initialize
+        :param default_value : Value applied if the attribute is not in the dict
+        """
+        if attribute_name in att_dict.keys():
+            value = att_dict[attribute_name]
+        else:
+            value = default_value
+
+        setattr(self, attribute_name, value)
+
+    def _verify_device_especification(self, dev_spec):
+        """Verifies a drive group device specification
+
+        Attributes supported in the device specification are:
+            {
+            id_model : A wildcard string. e.g: "SDD*"
+            size     : Size specification of format LOW:HIGH.
+                        Can also take the the form :HIGH, LOW:
+                        or an exact value (as ceph-volume inventory reports)
+            rotates  : is the drive rotating or not
+            count    : if this is present limit the number of drives to this
+                        number.
+            }
+
+        Any attributes (even none) can be included in the device
+        specification structure.
+
+        :param device_spec: Dict with the device specification desired
+        """
+        # A dict needed
+        if type(dev_spec) is not dict:
+                raise DriveGroupError("Device specification must be a \
+                                        dictionary")
+
+        # Only certain attributes are allowed:
+        allowed_att = frozenset(["id_model", "size", "rotates", "count"])
```

Comment:
1. why `frozenset`?
1. where does `id_model`, `count` and `rotates` come from?

Reply:
1. Because it is an immutable set
2. https://github.com/SUSE/DeepSea/wiki/Drive-Groups